src/os/posix/x11.rs

```diff
@@ -1024,8 +1049,25 @@ impl Window {
             return;
         }
 
+        let char = unsafe { self.code_point_to_char(&mut ev.key, code_point) };
         if let Some(ref mut callback) = self.key_handler.key_callback {
-            callback.add_char(code_point);
+            callback.add_char(char);
+        }
+    }
+
+    unsafe fn code_point_to_char(&mut self, event: &mut XKeyEvent, code_point: u32) -> u32{
+        const BUFFER_SIZE: usize = 32;
+
+        unsafe {
+            let mut buff: Box<[u8]> = vec![0; BUFFER_SIZE].into_boxed_slice(); //TODO: extra heap allocation
```

Comment:
Also here remove this heap allocation with something like ` let mut buffer: [u8; 32] = [0; 32];`

Reply:
This was on my list of things to get done, I just wanted to get it working first. Normally I would not do a PR until I have such things cleaned up, but I need help, hence the early PR.